src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -311,7 +313,7 @@ MockExpectedCall_c* withConstPointerParameters_c(const char* name, const void* v
 
 MockExpectedCall_c* withFunctionPointerParameters_c(const char* name, void (*value)())
 {
-    expectedCall = &expectedCall->withParameter(name, value);
+    expectedCall = &expectedCall->withParameter(name, (Cpp_Func*)value);
```

Comment:
Unfortunately, no. Of course I can remove the comment. But the only way to cast to "C" or "C++" that I know of is via typedef. As it is not normally a problem, C / C++ doesn't seem to provide another way to cast void (*)() C to void(*)() C++.